src/mon/MDSMonitor.cc

```diff
@@ -501,6 +502,7 @@ bool MDSMonitor::prepare_beacon(MMDSBeacon *m)
       dout(0) << "got beacon for MDS in STATE_STOPPING, ignoring requested state change"
 	       << dendl;
       _note_beacon(m);
+      m->put();
```

Comment:
`prepare_update()` is responsible to free the message before returns. so this change is correct.